src/nodewatcher/nodewatcher.py

```diff
@@ -295,8 +317,16 @@ def _poll_instance_status(config, scheduler_module, asg_name, hostname, instance
 
     idletime = _init_idletime()
     instance_properties = get_instance_properties(config.region, config.proxy_config, instance_type)
+    start_time = None
     while True:
-        time.sleep(60)
+        end_time = datetime.now()
+        if not start_time:
+            start_time = end_time
+        time_delta = (end_time - start_time).total_seconds()
```

Comment:
What about creating a common function to use in the three daemons?

Something like:
```python
def wait_something(loop_time=60, start_time=None):
    time_delta = (datetime.now() - start_time).total_seconds() if start_time else 0
    if time_delta < LOOP_TIME:
        time.sleep(LOOP_TIME - time_delta)
```

Reply:
Good idea! done!